### PR TITLE
Dynamic Runtime

### PR DESCRIPTION
**Pull request by codewithkyle**
This PR introduces a dynamic JIT version of the DjinnJS runtime.

Lighthouse testing (v6.1.0) reports an average performance rating of 87 while the statically imported runtime reports an average of 63.

Developers can load the new `djinn.mjs` file directly OR they can add a fallback for browsers that don't support the dynamic import syntax.

```javascript
try{
    import(`${location.origin}/assets/djinn.mjs`);
}catch(error){
    document.write('<script type="module" src="/assets/runtime.mjs"><\/script>');
}
```

### CODE REVIEW

**Comment by codewithkyle** [comment]
### Test 1
Additional testing has shown that the performance score received with the dynamic runtime (87) is comparable to loading one main CSS file or inlining all the CSS with no JavaScript.

### Test 2
Testing also shows that my performance score was capped at 88 when loading a website without any CSS or JavaScript, suggesting that ~12 points are lost to the server & templating engine.

### Conclusion
Based on this information, it seems as though the dynamic runtime only slightly hinders the performance score and, any future performance optimizations will need to be server-based.

**Comment by codewithkyle** [comment]
I've updated this PR to include a bootstrap file. It handles Intersection Observer & Web Components polyfills and loads the dynamic runtime on any browser that supports [dynamic imports](https://v8.dev/features/dynamic-import) while using the statically imported runtime as a fallback.

File: src/core/dynamic-runtime.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        /** Scroll to top of page */
+        window.scrollTo(0, 0);
+    }
+
+    private async handleEagerResponse(files) {
+        const loadingMessage = document.body.querySelector("djinnjs-file-loading-value") || null;
+        if (document.documentElement.getAttribute("state") === "hard-loading" && this.loadingMessage) {
+            this.loadingMessage.setAttribute("state", "3");
+            this.loadingMessage.innerHTML = `Loading resources:`;
+            if (loadingMessage && usePercentage) {
```
Comment (bamccaul-dev) [inline src/core/dynamic-runtime.ts:121]: Let's make this a hook for the developer to be able to put in what they might need.

File: cli/lib/default-config.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ module.exports = {
     outDir: "assets",
     noCachePattern: /(\.json)$|(cachebust\.js)/gi,
     cachebustURL: "/cachebust.json",
+    resourcePattern: /(\.js)$|(\.css)$|(\.mjs)$|(\.cjs)$|(\.png)$|(\.jpg)$|(\.gif)$|(\.webp)$|(\.jpeg)$|(\.svg)$/gi,
```
Comment (bamccaul-dev) [inline cli/lib/default-config.js:7]: Can we also add a required connection default here that would consumed through the files?

**Review by bamccaul-dev — APPROVED** [review]

**Pull request merged**